src/hooks/useNetworkCheck.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import { ToastId } from 'react-toastify'
+import { toast } from 'toastify'
+
+import { useWalletConnection } from 'hooks/useWalletConnection'
+import useSafeState from 'hooks/useSafeState'
+
+import { Network } from 'types'
+
+const availableNetworks = CONFIG.exchangeContractConfig.config.map(({ networkId }) => networkId)
+
+const NetworkErrorMessage: React.FC = () => (
+  <div>
+    <div>Please switch to a supported network!</div>
+    <strong>{`${availableNetworks.map(
+      (id, idx, arr) => (idx === arr.length - 1 ? ' or ' : ' ') + Network[id],
+    )}`}</strong>
+    <br />
+    <br />
+    <div>
+      Click{' '}
+      <a href="https://github.com/gnosis/dex-react/wiki#-compatible-networks" target="_blank" rel="noreferrer">
+        here
+      </a>{' '}
+      for more information
+    </div>
+  </div>
+)
+
+function runCheckOnValidNetworks(networkId?: number): boolean {
+  if (!networkId) return true
+
+  return availableNetworks.includes(networkId)
```

Comment:
better make a Set for `availableNetworks`, then won't even need `useMemo` as lookup is O(1)
Not that you need useMemo for `<short_array>.includes`

Reply:
true true, not sure why i didnt think of that! :+1: